src/posts/2020-07-30-the-worst-decision-of-my-career.md

```diff
@@ -0,0 +1,271 @@
+reason about. Bigger teams with well enough defined scopes will benefit more
+from this.
+
+### Accounting
+
+For event-sourcing, the analogy is usually around accounting. Accountants don't
+use erasers, amiright? For thousands of years, we invented and evolved
+mechanisms to keep track of things. Double-entry accounting was invented to
+guarantee accurate financial records because it reveals data entry errors.
+_This is awesome, so let's do the same thing in code now! Besides, if you never
+delete anything, you can always go back and look, and maybe use that data for
+other purposes._
+
+Slow down! Accountants invented systems around manual work. Those systems were
+designed to prevent human error. We still have to account for human error in
+software, but it's a different kind of error, so we need a different kind of
+solution.
+
+Event-sourcing indeed solves a lot of problems, but there are other, simpler,
+solutions that you can use. You can keep track of the changes in a table (on
+your database) with just another table and some triggers. If you're into Rails,
+[paper_trail](https://github.com/paper-trail-gem/paper_trail) is the tool for
+you.
+
+Is your data time-series? No problem. Put those events to a table and use
+a materialized view to read the data. Small steps can take you a long way.
+
+### Banking
+
+If you happen to have event-sourcing enthusiasts in your team, they will jump
+to the opportunity to implement a payment system because banks use
+event-sourcing! But do they really? I have been looking for years, and no one
+that I've met could endorse this information with confidence. Maybe banks just
+use the same solution with a materialized view I mentioned above? Besides, why
+are we even talking about banks? This isn't a bank, and if it were a bank, it's
+just getting started. Banks didn't start as they are; they had to evolve into
+what you see.
+
+I know a few people who use event-sourcing in production in what I consider
+a proper use; most times, it's around the payment system. But these people work
+at companies with a well-established business and the infrastructure to support
+it.
+
+## Making the right call
+
+Am I going to use event-sourcing or microservices again? Sure, it depends on
+the context and the client. The decisions I make are the best that I can with
+the information that I have. For instance, when I'm part of the team that's
+starting a product, the technology we pick will depend on how we'll hire: if
+you want to build an office in Braga, we have to make sure we have developers
+for that technology available. We have to think things through, and some things
+you only learn from experience. This applies to the systems we design as well.
+I've seen enough people design around what they believe the product will become
+in two years to know that those designs always fail to accommodate the changes
+that will come. So I design systems for small, incremental changes. Do the
```

Comment:
If this is to be published in the Subvisual blog I think you can/should change this to "we". These are principles that I believe everyone shares and also a good lead to a "Hire us" call to action.

Reply:
👍